src/core/Akka.Tests/Pattern/CircuitBreakerStressSpec.cs

```diff
@@ -0,0 +1,134 @@
+            public int CircCount { get; }
+
+            public Result(int doneCount, int timeoutCount, int failCount, int circCount)
+            {
+                DoneCount = doneCount;
+                TimeoutCount = timeoutCount;
+                FailCount = failCount;
+                CircCount = circCount;
+            }
+        }
+
+        internal class StressActor : UntypedActor
+        {
+            private readonly CircuitBreaker _breaker;
+            private int _doneCount;
+            private int _timeoutCount;
+            private int _failCount;
+            private int _circCount;
+
+            public StressActor(CircuitBreaker breaker) => _breaker = breaker;
+
+            protected override void OnReceive(object message)
+            {
+                switch (message)
+                {
+                    case RequestJob _:
+                        _breaker.WithCircuitBreaker(Job).PipeTo(Self);
+                        break;
+                    case JobDone _:
+                        _doneCount++;
+                        break;
+                    case Status.Failure failure when failure.Cause is OpenCircuitException:
+                        _circCount++;
+                        _breaker.WithCircuitBreaker(Job).PipeTo(Self);
+                        break;
+                    case Status.Failure failure when failure.Cause is TimeoutException:
+                        _timeoutCount++;
+                        _breaker.WithCircuitBreaker(Job).PipeTo(Self);
+                        break;
+                    case Status.Failure _:
+                        _failCount++;
+                        _breaker.WithCircuitBreaker(Job).PipeTo(Self);
+                        break;
+                    case GetResult _:
+                        Sender.Tell(new Result(_doneCount, _timeoutCount, _failCount, _circCount));
+                        break;
+                    default:
+                        base.Unhandled(message);
+                        break;
+                }
+            }
+
+            private async Task<JobDone> Job()
+            {
+                await Task.Delay(TimeSpan.FromMilliseconds(300));
+                return JobDone.Instance;
+            }
+        }
+
+        public CircuitBreakerStressSpec(ITestOutputHelper output) 
+            : base(output)
+        { }
+
+        [Fact]
+        public void A_CircuitBreaker_stress_test()
+        {
+            var breaker = new CircuitBreaker(Sys.Scheduler, 5, TimeSpan.FromMilliseconds(200), TimeSpan.FromSeconds(200));
+            var stressActors = Enumerable.Range(0, 3).Select(i => Sys.ActorOf(Props.Create<StressActor>(breaker))).ToList();
+
+            for (var i = 0; i < 1000; i++)
+                foreach (var stressActor in stressActors)
+                {
+                    stressActor.Tell(RequestJob.Instance);
+                }
+
+            // let them work for a while
+            Thread.Sleep(3000);
```

Comment:
Nitpick - better to call `await Task.Delay` here. Has no semantic impact on the test but avoids taking a thread out of commission.